CMakeLists.txt

```diff
@@ -48,7 +48,7 @@ set(CMAKE_MACOSX_RPATH 1)
 
 # Enforce the C++ standard, and disable extensions
 if(NOT DEFINED CMAKE_CXX_STANDARD)
-    set(CMAKE_CXX_STANDARD 11)
+    set(CMAKE_CXX_STANDARD 14)
```

Comment:
We should take a look at this eventually to see if we can make this work with C++11 constexpr (for msvc's sake).  I think it's just a matter of inverting a few conditionals to be ternary ops.